app/models/endpoint.rb

```diff
@@ -41,6 +41,18 @@ def ssl_cert_store
     nil # use system defaults
   end
 
+  # Split concatenated PEM certs to a list
+  def certificate_authorities
+    return [] if certificate_authority.blank?
+    certificate_authority.split(/(?=-----BEGIN)/).reject(&:blank?)
```

Comment:
Oh, I completely missed the second part.
Currently leading text before first cert will be returned as part of the array and fail OpenSSL parsing.
However I now see text between certs, or after last cert `-----END CERTIFICATE-----`, is silently ignored by OpenSSL.

So what do you think we should do with extra texts: fail validation / allow and silently ignore when parsing / allow but silently strip on saving?